src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChainBuilder.java

```diff
@@ -340,6 +353,21 @@ public ItineraryListFilterChain build() {
       );
     }
 
+    // Filter transit itineraries by comparing against non-transit using generalized-cost
+    if (removeTransitWithHigherCostThanBestOnStreetOnly != null) {
+      filters.add(
+        new DeletionFlaggingFilter(
+          new RemoveTransitIfStreetOnlyIsBetterFilter(
+            removeTransitWithHigherCostThanBestOnStreetOnly
+          )
+        )
+      );
+    }
+
+    if (removeTransitWithMoreWalking) {
+      filters.add(new DeletionFlaggingFilter(new RemoveTransitWithMoreWalking()));
+    }
+
```

Comment:
I think both of these are "absolute filters" and should be moved inside the block below - the definition of "absolute filters" is a bit unclear. The point is that these must be run after the group-by filters. To do it properly the limits should be calculated first, before any filters are applied. We relay here on the fact that the direct street results are likely to be kept - not removed by a group-by-filter.